misc/dbt-materialize/dbt/include/materialize/macros/adapters.sql

```diff
@@ -86,6 +86,40 @@
     {{ sql }}
 {%- endmacro %}
 
+{% macro materialize__create_source(relation, sql) -%}
+  {% set contract_config = config.get('contract') %}
+  {% if contract_config.enforced %}
+    {{exceptions.warn("Model contracts cannot be enforced for custom materializations (see dbt-core #7213)")}}
+  {%- endif %}
+
+  {%- set cluster = config.get('cluster', target.cluster) -%}
+
+  create source {{ relation }}
+  {% if cluster %}
+    in cluster {{ cluster }}
+  {% endif %}
+
+  {{ sql }}
+  ;
+  {%- endmacro %}
+
+{% macro materialize__create_sink(relation, sql) -%}
+  {% set contract_config = config.get('contract') %}
+  {% if contract_config.enforced %}
+    {{exceptions.warn("Model contracts cannot be enforced for custom materializations (see dbt-core #7213)")}}
+  {%- endif %}
+  
+  {%- set cluster = config.get('cluster', target.cluster) -%}
+
+  create sink {{ relation }}
+  {% if cluster %}
+    in cluster {{ cluster }}
+  {% endif %}
```

Comment:
...oh, but that approach would be awkward for sinks. 😞

Reply:
🫠

Reply:
I mean...adding it to the `source` materialization but not the sink one would make the DDL users write consistent, no?

Reply:
Ah, nevermind.